File: src/swerc2018/10940.py
# 10940 Throwing cards away II
# helper script to better understand the problem
import queue
for i in range(1, 201):
    l = queue.deque(range(1, i + 1))
    print()
    print(bin(i))
    while len(l) > 0:
        if len(l) == 1:
            k = l.popleft()
            print(bin(k))
            print(k)
        else:
            l.popleft()
        if len(l) > 0:
            l.append(l.popleft())

File: src/swerc2018/10940.cc
// 10940 Throwing cards away II

#include <iostream>
using namespace std;
int log(int n) {
    for (int i = 0; ; i++) {
        if (1 << i > n) return (1 << i);
    }
}
int main() {
    int n;
    ios::sync_with_stdio(false);
    while (cin >> n and n) {
        if (n == 1) {
            cout << 1 << '\n';
        }
        else if (n == 2) {
            cout << 1 << '\n';            
        }
        else {
            cout << ((n << 1) % log(n)) << '\n';
        }
    }
}